### ISSUE TITLE
Checking target MCU ... stuck

### ISSUE THREAD

**Issue by Sliicy** [tomazas/ledcube8x8x8#8]
![image](https://user-images.githubusercontent.com/23116873/127087645-a2db071f-f6db-4f90-9ccd-fa756d24a668.png)
Hello @tomazas!
I assembled an iCubeSmart 8x8x8 cube from Amazon. It has the same chip type as in the tutorial (STC12C5A60S2).
Everything worked after assembly, so I then wanted to be able to control it from USB terminal.
I tried following the tutorial, but when I try to Download or Re-Program, it gets stuck on "Checking target MCU ...", and the CH340 USB Serial just flickers trying to flash.
I also tried both RX to TX and RX to RX, but it won't recognize the cube.
I also have the correct drivers installed:
![image](https://user-images.githubusercontent.com/23116873/127088143-4418f467-a97a-4b53-a1c2-fec9b26080fe.png)

Do you know what I'm doing wrong?

**Comment by Sliicy** [comment]
Also, the CH340 that shipped with this came with a jumper connecting RX and TX. I assumed that I needed to remove the jumper to be able to talk to it, right?
The pin-out on the CH340 is: 5V, VCC, 3V3, TXD, RXD, GND. I connected the 5V to the 5V on the Cube, ground to ground, and crossing RX and TX.

**Comment by Sliicy** [comment]
After further investigating, I was able to download the firmware finally!
However, now the cube just remains all brightly lit, not animating. When I try using the other Java program to control it, some LEDs flicker on certain animations, but it seems lost in translation. The only time it somewhat animates is on 19200 Baud. The other 2 programs (3D8 Win & 3D8S Alpha) somewhat make the lights flicker.
What I did:
I connected GND to GND, TXD to TXD, RXD to RXD, and left 5V unplugged. Program would not detect it otherwise.
I then changed the max and min baud to 9600.
With the device only plugged in via the CH340, I started the Download. As it was waiting for the MCU, I plugged in the cube to a separate power source.
And then it worked:
![image](https://user-images.githubusercontent.com/23116873/127097458-40155d32-88da-4519-a718-3c50a148ca29.png)

Here is a log of events:
```
Checking target MCU ... 
  MCU type: STC12C5A60S2
  F/W version: 7.1.4I

Current H/W Option:
  . Current system clock source is external crystal oscillator
  . Current clock frequency: 23.846MHz
  . Oscillator gain is HIGH
  . RESET pin behaves as reset pin
  . No low level detection on RESET2 pin
  . Power-on reset, use the extra power-on delay
  . Hardware do not enable Watch-Dog-Timer
  . Watch-Dog-Timer pre-scalar : 256
  . Watch-Dog-Timer stop count in idle mode
  . Program can modify the Watch-Dog-Timer scalar
  . Do not detect the level of P1.0 and P1.1 next download
  . Do not erase user EEPROM area at next download

  MCU type: STC12C5A60S2
  F/W version: 7.1.4I


Re-handshaking ... Successful			[0.484"]
Current Baudrate: 9600
Erasing MCU flash ...  OK !		[2.812"]
MCU ID : D17EC59205195F
Programming user code ... OK !		[6.516"]
Programming OPTIONS ... OK !		[0.093"]

H/W Option upgrade to:
  . Current system clock source is external crystal oscillator
  . Current clock frequency: 23.846MHz
  . Oscillator gain is HIGH
  . RESET pin behaves as reset pin
  . No low level detection on RESET2 pin
  . Power-on reset, use the extra power-on delay
  . Hardware do not enable Watch-Dog-Timer
  . Watch-Dog-Timer pre-scalar : 256
  . Watch-Dog-Timer stop count in idle mode
  . Program can modify the Watch-Dog-Timer scalar
  . Do not detect the level of P1.0 and P1.1 next download
  . Do not erase user EEPROM area at next download
MCU ID : D17EC59205195F

  MCU type: STC12C5A60S2
  F/W version: 7.1.4I


  Complete !(2021-07-27 00:49:00)
```

**Comment by Sliicy** [comment]
After further investigating, it seems that the lights are working with the Java program, albeit everything is controlled a line at a time (not individually as pixels). If I continue to have issues, I will open a new bug for dimensional drawing.

**Comment by tomazas** [comment]
> I tried following the tutorial, but when I try to Download or Re-Program, it gets stuck on "Checking target MCU ...", and the CH340 USB Serial just flickers trying to flash.

Usually you need to turn the cube on/off/reset the power for the detection to work. I think this is what happened when you plugged in the other power source and it worked.

> After further investigating, it seems that the lights are working with the Java program, albeit everything is controlled a line at a time (not individually as pixels). If I continue to have issues, I will open a new bug for dimensional drawing.

Please open only if you find a problem in software/firmware and you are confident that it's a bug (have steps to reproduce it properly).

**Comment by CityArchitect** [comment]
Hello Tomazas,

I built one of these things about  a year ago, getting it from ebay.

I put it all together ok, after a few mistakes, but it took me quite some
time to realise it had no software programmed into the STC chip!

I had to get a chip programmer, and I managed to obtain the correct
software that runs the famous demo :-)

I have all the software (C and HEX) that makes this thing work if you have
the device to program it!

Steve
